Completed Challenge and Review Pages to Final Form

fru1tstand opened the pull request:
+ Added mock challenge question
+ Completed live syntax highlighting "within" textarea
+ Added review code commenting
+ Wired score metric assignment

fru1tstand committed:
Completed challenge template; Implemented live syntax highlighting

fru1tstand committed:
Merge branch 'review' into flow-finalize

fru1tstand committed:
Completed review commenting pulldown; Wired score assignment

fru1tstand committed:
Small dashboard stylistic fixing

fru1tstand committed:
Merge branch 'master' into flow-finalize

fru1tstand commented:
Finishes #29, and #31

fru1tstand committed:
Added feedback page and wired it to dashboard

jonathanpli commented on src/main/resources/templates/challenge.html:
```diff
@@ -15,12 +15,185 @@
+	<div class="right">
+		<div>
+			<textarea class="code-highlight" id="challenge-input">import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
```
Not sure how to request a change for this, but when looking at the front end, if I try interacting with the code, a bunch of the code in the challenge shows up with red underline, even though the package imports look correct. Would you happen to know how to fix that?

fru1tstand replied:
Yeah, it's the browser's autocorrect. Good catch.

jonathanpli replied:
Cool! Seemed to fix most of the problems. There are still some red underlining, but here's an interesting one -- 
<img width="1092" alt="screen shot 2017-05-25 at 1 58 56 am" src="https://cloud.githubusercontent.com/assets/13812345/26443557/c6bc6eb0-40ed-11e7-97bb-2298c6e62415.png">

So there's the red underline, but if I try deleting one character, this is what happens --
<img width="1076" alt="screen shot 2017-05-25 at 1 59 53 am" src="https://cloud.githubusercontent.com/assets/13812345/26443579/e201f4ba-40ed-11e7-8498-0e3543aa7b2d.png">

EDIT: It looks like the indentation is just incorrect for all the code--the caret doesn't match where the input displays

EDIT 2: This problem only occurs on smaller screens; it works fine on my larger monitor

fru1tstand replied:
I didn't push to the server, but can you highlight the part that's underlined and screenshot it again? The way it's set up is a <textarea> is laying over the \<pre>\<code> block and the styles between the two should match exactly. The textarea is providing the ability to edit, show a caret, and highlight (not color, think copy/paste type of highlight) text, and the pre/code block is there to provide the syntax highlighting. The underlining of what appears to be blank space may be caused by the two not aligning properly.

jonathanpli replied:
Here's the text alignment problem--the red underline is gone now, interestingly enough.

<img width="447" alt="screen shot 2017-05-25 at 2 37 22 am" src="https://cloud.githubusercontent.com/assets/13812345/26444914/20196940-40f3-11e7-9fd4-3c2412f5f4f0.png">

fru1tstand replied:
I pushed to the server which removed all underlining. I also fixed the text collapse on all major browsers except IE. IE will never stop softwrapping the textarea which is throwing the alignment off. For now, I don't think I want to invest any more time into the issue as there are more important things than getting IE compliance. Let me know if any more issues arise in Firefox, Chrome, or Safari.

jonathanpli replied:
All of the underlining is gone (yay!), but the text alignment issue is still there for my smaller screen on Chrome, even after doing a hard cache refresh. Assuming we'll be using a larger monitor for the demo, and it doesn't affect the demo, though, I think it's fine to ignore the problem for now.

![image](https://cloud.githubusercontent.com/assets/13812345/26491508/cb06cd4e-41c4-11e7-9153-a27dd530809d.png)

fru1tstand committed:
Disabled browser autocorrect

fru1tstand committed:
Merge branch 'flow-finalize' into feedback

fru1tstand committed:
Fixed FF/IE Bug where wrapping didn't occur for linebreaks; Set challenge sidebar as fixed width

fru1tstand committed:
Merge branch 'flow-finalize' into feedback

fru1tstand committed:
First draft of earnings page

fru1tstand committed:
Changed layout of earnings and scaled down fonts and cards

fru1tstand committed:
Scaled down navbar sizes

fru1tstand committed:
Narrowed sidebar, scaled down fonts and spacing

fru1tstand committed:
Abstracted 'card' style; Added chartjs to dashboard; Scaled down dashboard

fru1tstand committed:
Fixed link colors on dashboard

fru1tstand committed:
Slimmed down sidebar for challenge, eval, and feedback; Scaled fonts,
spacing, and sizing to better fit; Abstracted review split screen to
code-split in layout scss file

fru1tstand committed:
Fixed li-sidebar active tab; Moved li-sidebar-nav-element from partial template to macro inside li-sidebar-page template

fru1tstand committed:
Removed page-specific partial templates in favor of macros

fru1tstand committed:
Reduced global nav link size

fru1tstand committed:
Push to server

fru1tstand committed:
Fixed box shadow layering

fru1tstand committed:
Fixed nav spacing; Code cleanup

fru1tstand committed:
Changed graph order to chronological from reverse chronological

fru1tstand committed:
Merge branch 'earnings' into quickfix-cleanups

fru1tstand committed:
Merge pull request #47 from DumplingLTD/quickfix-cleanups

Site-wide Resizing and Code Cleanup

fru1tstand committed:
Merge pull request #46 from DumplingLTD/earnings

Added Earnings Page

fru1tstand committed:
Merge pull request #43 from DumplingLTD/feedback

Added feedback page and wired it to the dashboard
